spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/S3Operations.java

```diff
@@ -103,4 +105,29 @@ default S3Resource upload(String bucketName, String key, InputStream inputStream
 	 * @return downloaded object represented as {@link S3Resource}
 	 */
 	S3Resource download(String bucketName, String key);
+
+	/**
+	 * Creates a signed URL for retrieving an object from S3.
+	 *
+	 * @param bucketName - the bucket name
+	 * @param key - the object key
+	 * @param duration - duration that the URL will work
+	 * @return a {@link URL} representing the signed URL
+	 */
+	URL createSignedGetURL(String bucketName, String key, Duration duration);
```

Comment:
Should we return signedHeaders, SignedPayload, and isBrowserExecutable as well here?

Since in AWS docs for signedHeaders -> ```Returns the subset of headers that were signed, and MUST be included in the presigned request to prevent the request from failing.```

Reply:
Now I am quite confident we need an end to end test for that.